tests/spec/localhost/nvidia_spec.rb

```diff
@@ -0,0 +1,13 @@
+require 'spec_helper'
+
+has_nvidiactl = file('/dev/nvidiactl').exists?
+
+# should always be installed
+describe file('/usr/bin/nvidia-docker') do
+  it { should exist }
+end
+
+# we just can test this on gpu enabled instances.
+describe command('nvidia-docker run --rm nvidia/cuda nvidia-smi'), :if => has_nvidiactl do
+  its(:stdout) { should contain /Tesla K80/ }
```

Comment:
Yes, you're right. Thinking about it again, this would add too much complexity without adding any additional value. Let's leave it like it is.